### PR TITLE
[pkg] rotate log file

### PR DESCRIPTION
**Pull request by huof6829**
# Description
Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes https://github.com/iotexproject/iotex-core/issues/2661

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by huof6829** [comment]
After reach `MaxSize` MB, s.log will be rotated auto.  After `MaxAge` days it will be removed auto.
The rotated log's name has the created time like :    /var/log/s-2022-09-14T19-27-26.648.log

**Comment by Liuhaai** [comment]
pls fix the log rotate issue with [logrotate](https://linux.die.net/man/8/logrotate)

**Comment by huof6829** [comment]
> pls fix the log rotate issue with [logrotate](https://linux.die.net/man/8/logrotate)

 It's a linux tool. cannot use in win/mac

File: pkg/log/log.go [hunk 1]
```diff
@@ -15,16 +15,18 @@ import (
...
 )
 
 // GlobalConfig defines the global logger configurations.
 type GlobalConfig struct {
-	Zap                *zap.Config `json:"zap" yaml:"zap"`
-	StderrRedirectFile *string     `json:"stderrRedirectFile" yaml:"stderrRedirectFile"`
-	RedirectStdLog     bool        `json:"stdLogRedirect" yaml:"stdLogRedirect"`
-	EcsIntegration     bool        `json:"ecsIntegration" yaml:"ecsIntegration"`
+	Zap                *zap.Config        `json:"zap" yaml:"zap"`
+	Lumberjack         *lumberjack.Logger `json:"lumberjack" yaml:"lumberjack"`
```
Comment (dustinxie) [inline pkg/log/log.go:26]: move to end

File: config/standalone-config.yaml [hunk 2]
```diff
@@ -15,4 +15,9 @@ log:
     outputPaths: ["stderr", "stdout"]
     errorOutputPaths: ["stderr"]
   stderrRedirectFile: /var/log/s.log
-  stdLogRedirect: true
\ No newline at end of file
+  stdLogRedirect: true
+  lumberjack:
```
Comment (dustinxie) [inline config/standalone-config.yaml:19]: is this included in Alpine image?

Comment (dustinxie) [reply config/standalone-config.yaml:19]: or test if logrotate can work or not

**Comment by huof6829** [comment]
`crond` service in container cannot autorun after `docker run`,  `logrotate` depends on `crond` and cannot effect also.  I config `crond` autorun and correct log date-format. I test it in local, log rotate successfully. the rotate-log is like : ` /var/log/s-2022-09-20.log`

**Comment by CoderZhi** [comment]
@xianhuawei @huof6829 could you try this PR in k8s?

**Comment by huof6829** [comment]
> @xianhuawei @huof6829 could you try this PR in k8s?

ok, we will run in testnet

**Comment by huof6829** [comment]
Run smoothly in testnet. 
WechatIMG43

**Review by Liuhaai — APPROVED** [review]

File: entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+/usr/sbin/crond
+
+exec "$@"
```
Comment (huangzhiran) [inline entrypoint.sh:5]: this mean crond run iotex-server?

**Review by huangzhiran — APPROVED** [review]

File: entrypoint.sh (hunk 3 above)
Comment (huof6829) [reply entrypoint.sh:5]: run crond then run  iotex-server

**Pull request merged**